pkg/client/block/client.go

```diff
@@ -58,20 +73,80 @@ type blockClient struct {
 	// These enable the EventsReplayClient to correctly map the raw event bytes
 	// to Block objects and to correctly return a BlockReplayObservable
 	eventsReplayClient client.EventsReplayClient[client.Block]
+	// latestBlockMu is a mutex that protects the latestBlock field from concurrent
+	// writes from getLatestBlocks and getInitialBlock.
+	latestBlockMu *sync.Mutex
+	// latestBlock is the last block observed by the blockClient.
+	// It is initialized by the getInitialBlock function and then updated by the
+	// updateLatestBlock goroutine that listens to the eventsReplayClient.
+	latestBlock client.Block
 }
 
 // CommittedBlocksSequence returns a replay observable of new block events.
 func (b *blockClient) CommittedBlocksSequence(ctx context.Context) client.BlockReplayObservable {
 	return b.eventsReplayClient.EventsSequence(ctx)
 }
 
-// LatestsNBlocks returns the last n blocks observed by the BockClient.
+// LastNBlocks returns the last n blocks observed by the BlockClient.
 func (b *blockClient) LastNBlocks(ctx context.Context, n int) []client.Block {
 	return b.eventsReplayClient.LastNEvents(ctx, n)
 }
 
+// LastBlock returns the last blocks observed by the BlockClient.
+func (b *blockClient) LastBlock() client.Block {
+	b.latestBlockMu.Lock()
+	defer b.latestBlockMu.Unlock()
+	return b.latestBlock
+}
+
 // Close closes the underlying websocket connection for the EventsQueryClient
 // and closes all downstream connections.
 func (b *blockClient) Close() {
 	b.eventsReplayClient.Close()
 }
+
+// getLatestBlocks listens to the EventsReplayClient's EventsSequence and updates
+// the latestBlock field with the latest block observed.
+func (b *blockClient) getLatestBlocks(ctx context.Context) {
+	latestBlockCh := b.eventsReplayClient.EventsSequence(ctx).Subscribe(ctx).Ch()
+	for block := range latestBlockCh {
+		b.latestBlockMu.Lock()
+		b.latestBlock = block
+		b.latestBlockMu.Unlock()
+	}
+}
+
+// getInitialBlock fetches the initial block from the chain and sets it as the
+// latest block observed by the blockClient.
+// This is necessary to ensure that the blockClient has the latest block when
+// it is first created.
+// If b.latestBlock is already set by the getLatestBlocks goroutine, then this
+// function skips setting the initial block.
```

Comment:
Why?

Reply:
If the subscription updates the latest block before the initial query, then there is no need to re-assign it again, since the subscription may come with a newer block (if the client comes with a late reply)